spec/file_reference_spec.rb

```diff
@@ -29,11 +29,30 @@
   end
 
   describe '#apply' do
-    it 'copies the source to the destination' do
-      allow(FileUtils).to receive(:mkdir_p)
-      allow(subject).to receive(:source_contents).and_return('source contents')
-      expect(File).to receive(:write).with(subject.destination_file_path, 'source contents')
-      subject.apply
+    context 'when the destination is not ignored in Git' do
+      it 'raises' do
+        allow(Fastlane::Helper::GitHelper).to receive(:is_ignored?)
+          .with(path: subject.destination_file_path)
+          .and_return(false)
+
+        expect(FileUtils).to_not receive(:mkdir_p)
+        expect(subject).to_not receive(:source_contents)
+        expect(File).to_not receive(:write)
+        expect { subject.apply }.to raise_error(StandardError)
+      end
+    end
+
+    context 'when the destination is ignored in Git' do
+      it 'copies the source to the destination' do
+        allow(Fastlane::Helper::GitHelper).to receive(:is_ignored?)
+          .with(path: subject.destination_file_path)
+          .and_return(true)
+
+        allow(FileUtils).to receive(:mkdir_p)
+        allow(subject).to receive(:source_contents).and_return('source contents')
+        expect(File).to receive(:write).with(subject.destination_file_path, 'source contents')
+        subject.apply
+      end
```

Comment:
I don't like how heavy on runtime mocking these tests are, they're more about the implementation (call `File.write`) than the actual behavior.

On the other hand, this was simpler to achieve than putting in place a temp folders etc.